eis_toolkit/spatial_analyses/cba.py

```diff
@@ -0,0 +1,511 @@
+# -*- coding: utf-8 -*-
+
+"""Created on Thu Jun  1 08:57:13 2023.
+
+@author: vella.
+"""
+
+from numbers import Number
+
+# os.environ['USE_PYGEOS'] = '0'
+from typing import Optional, Tuple, Union
+
+# import os
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import rasterio
+from shapely import wkt
+from shapely.geometry import Polygon
+
+
+def init_from_vector_data(
+    cell_size: int,
+    geodataframe: gpd.GeoDataFrame,
+    column: str = "",
+    subset_of_target_attribute_values: Optional[list] = None,
+) -> None:
+    """Creation of CBA object.
+
+    Initializes a CBA matrix from a vector file. The mesh is calculated
+    according to the geometries contained in this file and the size
+    of cells.
+
+    Args:
+        cell_size: size of the cells (ex : 3000).
+        geodataframe: path to vector file.
+        target_attribut: <optionnel> name of the attribute of interest.
+            If no attribute is specified, then an artificial attribute is created
+            representing the presence or absence of the geometries of this file for
+            each cell of the CBA grid. If a target attribute is indicated, it must be
+            categorical. Other types of attributes (numerical/string) are not managed.
+            A categorical attribute will generate as many columns (binary) in the
+            CBA matrix than values considered of interest
+            (dummification). See parameter <subset_of_target_attribut_values>
+        subset_of_target_attribut_values: <optionnel> list of values
+            of interest of the target attribute, in case a categorical target attribute
+            has been specified. Allows to filter a subset of relevant values.
+
+    Returns:
+        CBA object is initialized.
+    """
+
+    # Allows a non-mutable default value
+    if subset_of_target_attribute_values is None:
+        subset_of_target_attribute_values = []
+
+    # Reading the vector file
+    geodata = geodataframe.copy(deep=True)
+
+    # Initialization of the grid
+    grid = get_grid(geodata, cell_size, cell_size)
+    grid.crs = geodata.crs
+
+    # Filling the grid with overlapped geometries
+    indicators, join_grid = prepare_grid(geodataframe, grid, column, subset_of_target_attribute_values)
+    tmp = join_grid[["geom"]].copy(deep=True)
+    tmp.rename({"geom": "geometry"}, axis=1, inplace=True)
+
+    # Saving results in attributes
+    cba = gpd.GeoDataFrame(pd.concat([tmp.groupby(tmp.index).first(), indicators], axis=1))
```

Comment:
Right now the Python module is `cba.py` and within the code `cba` is used as the variable name for the `GeoDataFrame` holding the data. This can cause mistakes when the module is used. I would rename either the module (e.g. `cba.py` -> `cell_based_association.py`, I prefer this option!) or the `cba` variable in code (e.g. `cba` -> `cba_data`).

Reply:
I renamed both `cba.py` and `cba_test.py` to `cell_based_association.py` and `cell_based_association_test.py` respectively. As you raised, it should be more coherent than having duplicate names.